### ISSUE TITLE
What do I need to run on server?

### ISSUE THREAD

**Issue by sunglee** [davezuko/react-redux-starter-kit#613]
When I run `npm start` command and open browser with `localhost:3000` shows example site flawlessly. However, when I run `npm run deploy` and upload to a remote server, the page shows `<noscript />` tag. Should I setup anything on the remote server to run this?

I am pretty sure this is a great starter kit, but since it doesn't work on remote, I cannot go further.
Thanks for your help.

**Comment by davezuko** [comment]
You should just be able to point your web server at the `dist` folder and re-route any non-index routes to the root `index.html` file. It's not recommended, but you _should_ be able to run:

```bash
NODE_ENV=production npm start
```

Which will start the Koa server without any of the webpack middleware but it will serve the compiled `dist` folder. If _that_ isn't working then there is indeed an error somewhere.

**Comment by sunglee** [comment]
So I need to run Koa server on the remote? Is it possible to use this starter kit w/o remote (or local) Koa server?

**Comment by davezuko** [comment]
You do not need to run an app server (Koa, Express) for the app to work. You _do_ need some sort of web server to serve the static files (Nginx) that are compiled to `dist`, but that's all.

Here's some info about using other local servers: https://github.com/davezuko/react-redux-starter-kit/issues/581

**Comment by sunglee** [comment]
So you need something in the server such as Nginix to make `dist` work on remote server?

**Comment by davezuko** [comment]
You can use whatever you want, Apache, Nginx, anything. All you have to do is point all incoming route requests to your compiled `/dist` folder, so that the `index.html` in the root of the directory is served.

Unfortunately I'm very weak with Apache, but I imagine there's a fairly easy way to do this with the `.htaccess` file. If you'd prefer you could spin up the koa server (`NODE_ENV=production npm start`) and then point all incoming requests to `localhost:3000` and that _also_ should work, it will just be a bit slower than serving static files with a web server.

**Comment by Neuw84** [comment]
Could anyone put an example on running the compiled `/dist` folder on nginx of apache? 

In nginx if I use `try_files $uri$args $uri$args/ /index.html;` for redirecting to the root page the web seems to be working ok. However when pressing F5 the web content is found but is not rendered. 

That's it using Chrome developers tools it can be seen that the contents are loading correctly (If I do not put that redirection on nginx I only get a 404) but justs renders blank page. 

If I run the `sudo npm run dev:no-debug` (sudo for binding to 80 port) it works well, but that's not production mode.

Someone has succeded?

**Comment by sunglee** [comment]
Thanks for all your help. I finally figured it out. The problems that I had was I did put the outputs from `/dist` folder in subfolder inside my server; for example, I put all exported files under `mydomain.com/example/app` so the react-router didn't know the basepath.

All I had to do was running `BASENAME='/example/app' npm run deploy` command and react-router will set the base path properly.

Thanks again for your kind help and time.
